src/data/polynomial/basic.lean

```diff
@@ -33,30 +33,29 @@ variables {R : Type u} {a : R} {m n : ℕ}
 section semiring
 variables [semiring R] {p q : polynomial R}
 
-instance : inhabited (polynomial R) := finsupp.inhabited
+instance : inhabited (polynomial R) := add_monoid_algebra.inhabited _ _
 instance : semiring (polynomial R) := add_monoid_algebra.semiring
 instance : has_scalar R (polynomial R) := add_monoid_algebra.has_scalar
 instance : semimodule R (polynomial R) := add_monoid_algebra.semimodule
 
-instance subsingleton [subsingleton R] : subsingleton (polynomial R) :=
-⟨λ _ _, ext (λ _, subsingleton.elim _ _)⟩
+instance [subsingleton R] : unique (polynomial R) := add_monoid_algebra.unique
 
-@[simp] lemma support_zero : (0 : polynomial R).support = ∅ := rfl
+@[simp] lemma support_zero : (0 : polynomial R).support = ∅ := finsupp.support_zero
```

Comment:
Is this type of change fine? While obviously this is still a `rfl` lemma, it seems helpful to the reader to show we're just exposing existing lemmas (and might make metaprogramming them out later easier).

Reply:
Does it still tagged as a `rfl` lemma after this change? If not, `dsimp` will no longer use it.

Reply:
I'll check.

Reply:
It's not, but I can tag it as such.

Reply:
Wow, things get very broken if I put `@[refl]` on this.

Reply:
```suggestion
@[simp] lemma support_zero : (0 : polynomial R).support = ∅ := rfl
```